sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc

```diff
@@ -0,0 +1,976 @@
+The `scratch` value must be the same for all work-items in `group`.
+|===
+
+Table 10. Member functions of the `group_with_scratchpad` class.
+|===
+|Member function|Description
+
+|`Group get_group() const`
+|Returns the `Group` class object that is handled by the `group_with_scratchpad` object.
+
+|`sycl::span<std::byte, Extent>
+get_memory() const`
+|Returns `sycl::span` that represents an additional memory
+that is handled by the `group_with_scratchpad` object.
+
+|===
+
+=== SYCL Properties for Interfaces with Fixed-size Private Arrays
+
+Group algorithms using the fixed-size array interface are performed across
+`N * ItemsPerWorkItem` elements in the group, where `N` is the work-group size and
+`ItemsPerWorkItem` is the number of elements that are processed by one work-item.
+
+When a work-item contributes multiple values to a group algorithm,
+there are multiple ways to interpret the order of that data.
+Let `r` is a virtual range for sorting of `N * ItemsPerWorkItem` elements.
+The extension supports two data placements:
+
+a) Data from the `[r + id * ItemsPerWorkItem; r + (id + 1) * ItemsPerWorkItem)` virtual range
+placed into the private memory under the span for `id`-th work-item.
+
+b) `i * N + id` element of `r` fill the `i`-th element of the private memory
+under the span for `id`-th work-item.
+
+To specify a correct data placement for placing of resulting data there is a enum:
+
+[source,c++]
+----
+class enum group_algorithm_data_placement{
+  blocked,
+  striped
+};
+----
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`
+to specify a data placement described in a).
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`
+to specify a data placement described in b).
+
+Example:
+
+N = 3;
+
+|===
+|Work-item id|Input private fixed-size array
+
+|0
+|{11, 10, 9, 8}
+|1
+|{7, 6, 5, 4}
+|2
+|{3, 2, 1, 0}
+|===
+
+After performing sorting by ascending there is the following virtual range:
+`{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}`.
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`.
+
+|===
+|Work-item id|Output private fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`.
+
+|===
+|Work-item id|Output private fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+There are 3 properties that satisfy
+link:SYCL_EXT_ONEAPI_PROPERTIES.asciidoc[SYCL Properties Extension] requirements:
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::property
+{
+    template<group_algorithm_data_placement type>
+    struct input_data_placement; // (1)
+
+    template<group_algorithm_data_placement type>
+    struct output_data_placement; // (2)
+}
+----
+
+1. `input_data_placement` specifies the data placement for input.

```

Comment:
Why is `input_data_placement` useful?  The input data isn't sorted, so it doesn't matter whether you interpret it as `blocked` vs. `striped`.

Reply:
It's important when you do stable sorting (where the order in original sequences is important) or use sorting algorithms that can handle pre-sorted sequences.

Reply:
OK, that is a good reason.  Please add a sentence explaining the reason.  For example:

>  `input_data_placement` specifies the data placement for input.  This is useful for stable sorts, which preserve the relative input order for elements that compare equal.